### ISSUE TITLE
BrowserWindow.getNativeWindowHandle() returns same value in different BrowserWindow

### ISSUE THREAD

**Issue by KochiyaOcean** [electron/electron#15911]
* Output of `node_modules/.bin/electron --version`: 4.0.0-beta.8
* Operating System (Platform and Version): Windows 10 1803
* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable): 4.0.0-beta.7

**Expected Behavior**
`BrowserWindow.getNativeWindowHandle()` should return different value for different window

**Actual behavior**
`BrowserWindow.getNativeWindowHandle()` returns same value for different window, which is not true nativeWindowHandle for any of them.

**To Reproduce**
Open two `BrowserWindow`, and run `getNativeWindowHandle` on both `BrowserWindow`
```js
const { remote } = require('electron')
w1 = new remote.BrowserWindow()
w2 = new remote.BrowserWindow()
console.log(w1.getNativeWindowHandle(), w2.getNativeWindowHandle())
```
![image](https://user-images.githubusercontent.com/8194131/49328509-7411a800-f5b5-11e8-9a24-827aa589d937.png)

**Comment by estoykov** [comment]
Probably it is caused by these fixes:
https://github.com/electron/electron/pull/15521
https://github.com/electron/electron/pull/15644
which implementation is wrong for both Windows and MacOS.
- for Windows it is wrong because the returned pointer to the HWND is somewhere in the stack, i.e. will point to some garbage at the time when it will be used
- for MacOS returned pointer is the direct pointer to the NSView, and the first 8 bytes from NSView internal structure will copied, not the pointer to the NSView

**Comment by estoykov** [comment]
This is still broken in 3.1.0, but fixed in 4.x.x. Can we backport the fix in 3.1.0?

**Comment by nikolovp** [comment]
I am also interested in cherry picking this fix into 3.*

**Comment by alexeykuzmin** [comment]
@estoykov @nikolovp 
The fix will be included in the next 3.1.x release.
